stablehlo/dialect/TypeInference.cpp

```diff
@@ -255,6 +255,57 @@ LogicalResult verifyQPerAxisScaleAndZeroPointConstraints(
   return success();
 }
 
+LogicalResult verifyReshapeOpQuantizationConstraints(
+    std::optional<Location> location, Type operandTy, Type resultTy) {
+  // reshape_c1, dynamic_reshape_c1
+  if (failed(verifyQPerTensorScaleAndZeroPointConstraints(location, operandTy,
+                                                          resultTy)))
+    return failure();
+
+  // reshape_c1, dynamic_reshape_c1
+  if (failed(verifyQPerAxisScaleAndZeroPointConstraints(location, operandTy,
+                                                        resultTy)))
+    return failure();
+
+  // reshape_c3
+  if (allQuantized<quant::UniformQuantizedPerAxisType>(operandTy, resultTy)) {
```

Comment:
Does this current logic allow for per-axis operand but not per-axis result? Is that expected, or is that handled elsewhere?

Reply:
op constraints do not allow `per-axis operand` and `not per-axis result`.  only `quantization_dimension` can differ.

```
(C1)
element_type(result) is given by:
element_type(operand), if !is_per_axis_quantized(operand).
element_type(operand) except that quantization_dimension(operand) and quantization_dimension(result) may differ, otherwise.

```

Reply:
>> Does this current logic allow for per-axis operand but not per-axis result?
That is not allowed.